lib/src/common/object.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:codable/core.dart';
+
+/// A [Codable] that can encode and decode standard Dart objects (Maps, Lists, etc.).
+class ObjectCodable implements Codable<Object?> {
+  const ObjectCodable();
+
+  @override
+  Object? decode(Decoder decoder) {
+    return switch (decoder.whatsNext()) {
+      DecodingType.keyed || DecodingType.mapped || DecodingType.map => _decodeMap(decoder),
+      DecodingType.list || DecodingType.iterated => _decodeList(decoder),
+      DecodingType.string => decoder.decodeString(),
+      DecodingType.int => decoder.decodeInt(),
+      DecodingType.double => decoder.decodeDouble(),
+      DecodingType.bool => decoder.decodeBool(),
+      DecodingType.nil => null,
+      _ => decoder.decodeObjectOrNull(),
+    };
+  }
+
+  Map<String, Object?> _decodeMap(Decoder decoder) {
+    final map = <String, Object?>{};
+    var keyed = decoder.decodeKeyed();
+    for (Object? key; (key = keyed.nextKey()) != null;) {
+      map[key.toString()] = decoder.decodeObject(using: this);
+    }
+    return map;
+  }
+
+  List<Object?> _decodeList(Decoder decoder) {
```

Comment:
Minor nit: might make more sense to accept `KeyedDecoder` and `IteratedDecoder` for these two methods.
Its more consistent with code elsewhere